cross-chain/arbitrum/hardhat.config.ts

```diff
@@ -0,0 +1,119 @@
+import type { HardhatUserConfig } from "hardhat/config"
+
+import "@keep-network/hardhat-helpers"
+import "@nomiclabs/hardhat-waffle"
+import "@nomiclabs/hardhat-etherscan"
+import "hardhat-gas-reporter"
+import "hardhat-contract-sizer"
+import "hardhat-deploy"
+import "@typechain/hardhat"
+import "hardhat-dependency-compiler"
+
+const config: HardhatUserConfig = {
+  solidity: {
+    compilers: [
+      {
+        version: "0.8.17",
+        settings: {
+          optimizer: {
+            enabled: true,
+            runs: 1000,
+          },
+        },
+      },
+    ],
+  },
+
+  paths: {
+    artifacts: "./build",
+  },
+
+  networks: {
+    hardhat: {
+      deploy: ["deploy_l2"],
+    },
+    goerli: {
+      url: process.env.CHAIN_L1_API_URL || "",
+      chainId: 5,
+      deploy: ["deploy_l1"],
+      accounts: process.env.ACCOUNTS_L1_PRIVATE_KEYS
+        ? process.env.ACCOUNTS_L1_PRIVATE_KEYS.split(",")
+        : undefined,
+      tags: ["etherscan"],
+    },
+    mainnet: {
+      url: process.env.CHAIN_L1_API_URL || "",
+      chainId: 1,
+      deploy: ["deploy_l1"],
+      accounts: process.env.ACCOUNTS_L1_PRIVATE_KEYS
+        ? process.env.ACCOUNTS_L1_PRIVATE_KEYS.split(",")
+        : undefined,
+      tags: ["etherscan"],
+    },
+    arbitrumGoerli: {
+      url: process.env.CHAIN_L2_API_URL || "",
+      chainId: 421613,
+      deploy: ["deploy_l2"],
+      accounts: process.env.ACCOUNTS_L2_PRIVATE_KEYS
+        ? process.env.ACCOUNTS_L2_PRIVATE_KEYS.split(",")
+        : undefined,
+      tags: ["arbiscan"],
+      companionNetworks: {
+        l1: "goerli",
+      },
+    },
+    arbitrumMainnet: {
+      url: process.env.CHAIN_L2_API_URL || "",
+      chainId: 42161,
+      deploy: ["deploy_l2"],
+      accounts: process.env.ACCOUNTS_L2_PRIVATE_KEYS
+        ? process.env.ACCOUNTS_L2_PRIVATE_KEYS.split(",")
+        : undefined,
+      tags: ["arbiscan"],
+      companionNetworks: {
+        l1: "mainnet",
+      },
+    },
+  },
+
+  external: {
+    deployments: {
+      goerli: ["./external/goerli"],
+      arbitrumGoerli: ["./external/arbitrumGoerli"],
+      mainnet: ["./external/mainnet"],
+      arbitrumMainnet: ["./external/arbitrumMainnet"],
+    },
+  },
+
+  etherscan: {
+    apiKey: {
+      goerli: process.env.ETHERSCAN_API_KEY,
+      arbitrumGoerli: process.env.ARBISCAN_API_KEY,
+    },
+  },
+
+  namedAccounts: {
+    deployer: {
+      default: 1,
+      goerli: process.env.CONTRACT_L1_OWNER_ADDRESS || "",
```

Comment:
If no environment variable is provided, what should happen?

Reply:
In the current flow, it won't be deployed. We can replace it with:
```
goerli: process.env.CONTRACT_L1_OWNER_ADDRESS || 0
```

Reply:
In other packages for goerli we use `0` for all the accounts. Don't we want to simplify it and do the same here instead of defining `CONTRACT_L1_OWNER_ADDRESS` and `CONTRACT_L2_OWNER_ADDRESS`?

Reply:
`CONTRACT_L1_OWNER_ADDRESS` and `CONTRACT_L2_OWNER_ADDRESS` are the accounts that have already some goerli ETH and you can use them to test deployment from your local box. This is how I tested the deployment scripts and I found it useful.

Reply:
d9ee3c33d76c1a110c5b8a098f60ec55f165d36d